weave/trace_server/file_storage.py

```diff
@@ -219,23 +253,40 @@ def store(self, uri: AzureFileStorageURI, data: bytes) -> None:
     @create_retry_decorator("azure_read")
     def read(self, uri: AzureFileStorageURI) -> bytes:
         """Read data from Azure container with automatic retries on failure."""
+        assert isinstance(uri, AzureFileStorageURI) and uri.to_uri_str().startswith(
+            self.base_uri.to_uri_str()
+        )
         client = self._get_client(uri.account)
         container_client = client.get_container_client(uri.container)
         blob_client = container_client.get_blob_client(uri.path)
         stream = blob_client.download_blob()
         return stream.readall()
 
 
-def _get_storage_client(uri: FileStorageURI) -> FileStorageClient:
+def get_storage_client_from_env() -> Optional[FileStorageClient]:
```

Comment:
slightly confusing name given that this is an optional

Reply:
Fixed